### PR TITLE
Complete parser

### PR DESCRIPTION
**Pull request by aasim**

### CODE REVIEW

File: test/Microsoft.Sbom.Api.Tests/Executors/ExternalDocumentReferenceWriterTest.cs [hunk 1]
```diff
@@ -76,13 +76,13 @@ await foreach (var result in results.ReadAllAsync())
 
                 Assert.IsNotNull(root);
 
-                if (root.TryGetProperty("SpdxDocument", out JsonElement documentNamespace))
```
Comment (aasim) [inline test/Microsoft.Sbom.Api.Tests/Executors/ExternalDocumentReferenceWriterTest.cs:79]: This is inside the api tests, so made it agnostic to the format.

File: src/Microsoft.Sbom.Parsers.Spdx22SbomParser/SPDXParser.cs [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+    {
+        Name = Constants.SPDXName,
+        Version = Constants.SPDXVersion
+    };
+
+    /// <inheritdoc/>
+    public SBOMMetadata GetMetadata()
+    {
+        throw new NotImplementedException();
+    }
```
Comment (edgarrs) [inline src/Microsoft.Sbom.Parsers.Spdx22SbomParser/SPDXParser.cs:106]: Do we need this method?

File: src/Microsoft.Sbom.Parsers.Spdx22SbomParser/SPDXParser.cs [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+                    result = ParserState.FINISHED;
+                }
+
+                isFinalBlock = reader.IsFinalBlock;
+                readerState = reader.CurrentState;
+                return result;
+            }
+            catch (JsonException e)
+            {
+                throw new ParserException($"Error while parsing JSON at position {stream.Position}, additional details: ${e.Message}", e);
```
Comment (edgarrs) [inline src/Microsoft.Sbom.Parsers.Spdx22SbomParser/SPDXParser.cs:157]: >stream [](http://example.com/codeflow?start=82&length=6)

Could this be null?

**Review by edgarrs — APPROVED** [review]
:shipit:

**Pull request merged**